cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/HttpClients.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.capability;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.common.http.DefaultHttpRequestConfig;
+import io.cdap.cdap.common.service.Retries;
+import io.cdap.cdap.common.service.RetryStrategies;
+import io.cdap.common.http.HttpRequest;
+import io.cdap.common.http.HttpRequests;
+import io.cdap.common.http.HttpResponse;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Makes {@link io.cdap.common.http.HttpRequest}s, retrying on failure.
+ */
+public final class HttpClients {
+  private static final int RETRY_LIMIT = 5;
+  private static final int RETRY_DELAY = 5;
+
+  private HttpClients() {
+  }
+
+  /**
+   * Makes an {@link HttpRequest}, retries on failure.
+   * @return {@link HttpResponse}
+   */
+  @VisibleForTesting
```

Comment:
Why all methods have this annotation? Generally public methods on util class is meant to be used by prod code, and the @VisibleForTesting annotation generally should only apply to package private methods.

Reply:
fixed